### PR TITLE
Add support for calculate and dynamically change offset position for …

### PR DESCRIPTION
**Pull request by romanlex**
Sorry for bad English.
Add support for calculate and dynamically change offset position for dropdown menu.
Function call always when call Popper.js modifierFn.
Now I can change dynamically position of dropdown menu 
```javascript
            $('[data-toggle="dropdown"]').dropdown({
                offsetFn: function(data) {
                    data.left = -100;
                    return data
                }
            });
```

### CODE REVIEW

**Comment by romanlex** [comment]
When use this code found bug in Popper.js(( Cannot set offset.left more when element width but can set any negative values

**Comment by romanlex** [comment]
I think what better send all popper params, because it's more customizable solution and can change more settings of Popper
```javascript
_getOffset(data) {
  const popperOffset = $.extend({}, data.offsets, this._config.offsetFn(data.offsets) || {})
  Object.keys(popperOffset).forEach((key) => {
    data.offsets[key] = popperOffset[key]
  })
  return data
}
```

File: dist/js/bootstrap.bundle.js [hunk 1]
```diff
@@ -3890,10 +3890,14 @@ var Dropdown = function () {
   };
   var Default = {
     offset: 0,
+    offsetFn: function offsetFn(data) {
```
Comment (Johann-S) [inline dist/js/bootstrap.bundle.js]: Please do not commit dist files, you must revert this change

File: docs/4.0/components/dropdowns.md [hunk 2]
```diff
@@ -593,6 +593,12 @@ Options can be passed via data attributes or JavaScript. For data attributes, ap
       <td>Offset of the dropdown relative to its target. For more information refer to Popper.js's <a href="https://popper.js.org/popper-documentation.html#modifiers..offset.offset">offset docs</a>.</td>
     </tr>
     <tr>
+      <td>offsetFn</td>
```
Comment (Johann-S) [inline docs/4.0/components/dropdowns.md]: No need for an new option here, just check if `offset` is a function or not

File: js/src/dropdown.js [hunk 3]
```diff
@@ -290,7 +301,7 @@ const Dropdown = (() => {
 
     static _clearMenus(event) {
       if (event && (event.which === RIGHT_MOUSE_BUTTON_WHICH ||
-        event.type === 'keyup' && event.which !== TAB_KEYCODE)) {
+          event.type === 'keyup' && event.which !== TAB_KEYCODE)) {
```
Comment (Johann-S) [inline js/src/dropdown.js]: you have unrelated changes here and their...

**Review by Johann-S — CHANGES_REQUESTED** [review]

**Comment by Johann-S** [comment]
Why did you closed your PR @romanlex ?

**Comment by romanlex** [comment]
@Johann-S big thx for your reply and thx for what you teach and say how to properly make changes.
I tried to correct your comments and made a new one PR

**Comment by Johann-S** [comment]
Ok :smiley: you're welcome :wink: but you can just update this one 
I made an issue related to that PR https://github.com/twbs/bootstrap/issues/24223

**Comment by romanlex** [comment]
I don't now ) How I can close this PR?

**Pull request closed without merging**